Instrument Host search, generalization of Context product access and Alpha MCP server

tloubrieu-jpl opened the pull request:
<!--
    ************************************** REMINDER **************************************
    PR Titles should be "user-friendly". We use these titles
    to populate our Release Notes. Some examples can be found here:
    https://github.com/NASA-PDS/nasa-pds.github.io/wiki/Issue-Tracking#pull-request-titles
-->

## 🗒️ Summary
- Generalization of the Context product access.
- New Instrument host search and enumaration
- Demo MCP server

## ⚙️ Test Data and/or Report
Unit test

Demo of MCP server in Claude described in README.

nutjob4life reviewed (changes requested):
It looks like these changes would revert the [Python 3.13 upgrade from 5 days ago](https://github.com/NASA-PDS/peppi/pull/105) back to Python 3.9:
```console
$ git checkout more_context
Switched to branch 'more_context'
Your branch is up to date with 'origin/more_context'.
$ egrep '3\.9' setup.cfg tox.ini 
setup.cfg:    Programming Language :: Python :: 3.9
setup.cfg:python_requires = >= 3.9
tox.ini:basepython = python3.9
$ echo uh oh
uh oh
$ 
```

Could you maybe re-base on a more recent `main` branch?
```console
$ git checkout main
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
$ egrep '3\.9' setup.cfg tox.ini 
$
```

tloubrieu-jpl commented:
Thanks @nutjob4life I merged main into the branch to integrate the upgrade to 3.13.
I also fixed most of the issues reported by Sonarcube and added a forgotten dependency.

nutjob4life reviewed (approved):
Looks great @tloubrieu-jpl, thanks!

Also:
```
$ tox
…
  py313: OK (265.62=setup[18.39]+cmd[247.23] seconds)
  docs: OK (27.89=setup[17.63]+cmd[10.25] seconds)
  lint: OK (3.27=setup[1.17]+cmd[2.10] seconds)
  congratulations :) (296.86 seconds)
```
